packages/context/src/__tests__/acceptance/class-level-bindings.acceptance.ts

```diff
@@ -145,34 +147,132 @@ describe('Context bindings - Injecting dependencies of classes', () => {
 
     expect(store.getter).to.be.Function();
-    expect(await store.getter()).to.equal('value');
+    expect(await store.getter()).to.equal('123');
 
     // rebind the value to verify that getter always returns a fresh value
-    ctx.bind('key').to('new-value');
-    expect(await store.getter()).to.equal('new-value');
+    ctx.bind('hash').to('456');
+    expect(await store.getter()).to.equal('456');
+  });
+
+  describe('in SINGLETON scope', () => {
+    it('throws if a getter cannot be resolved by the owning context', async () => {
+      class Store {
+        constructor(@inject.getter('hash') public getter: Getter<string>) {}
+      }
+
+      // Create a child context of `ctx`
+      const requestCtx = new Context(ctx, 'child');
+      // Bind `hash` to `value`
+      requestCtx.bind('hash').to('123');
+
+      // Bind `Store` as a singleton
+      ctx
+        .bind('store')
+        .toClass(Store)
+        .inScope(BindingScope.SINGLETON);
+
+      // Create the store instance from the child context
+      // Now the singleton references a `getter` to `binding`
+      const store = requestCtx.getSync<Store>('store');
+
+      await expect(store.getter()).to.be.rejectedWith(
+        /The key 'hash' is not bound to any value in context .+/,
+      );
+
+      // bind the value to to ctx
+      ctx.bind('hash').to('456');
+      expect(await store.getter()).to.equal('456');
+    });
+
+    it('throws if a value cannot be resolved by the owning context', async () => {
+      class Store {
+        constructor(@inject('hash') public hash: string) {}
+      }
+
+      // Create a child context of `ctx`
+      const requestCtx = new Context(ctx, 'child');
+      // Bind `hash` to `value`
+      requestCtx.bind('hash').to('123');
+
+      // Bind `Store` as a singleton
+      ctx
+        .bind('store')
+        .toClass(Store)
+        .inScope(BindingScope.SINGLETON);
+
+      await expect(requestCtx.get('store')).to.be.rejectedWith(
+        /The key 'hash' is not bound to any value in context .+/,
+      );
+
+      // bind the value to to ctx
+      ctx.bind('hash').to('456');
+      const store = await requestCtx.get<Store>('store');
+      expect(store.hash).to.equal('456');
+    });
+
+    it('injects method parameters from a context below the chain', async () => {
+      class Store {
+        private name = 'my-store';
+
+        static getHash(@inject('hash') hash: string) {
+          return hash;
+        }
+
+        getHashWithName(@inject('hash') hash: string) {
+          return `${hash}: ${this.name}`;
+        }
+      }
+
+      // Create a child context of `ctx`
+      const requestCtx = new Context(ctx, 'child');
+      // Bind `hash` to `value`
+      requestCtx.bind('hash').to('123');
+
+      // Bind `Store` as a singleton
+      ctx
+        .bind('store')
+        .toClass(Store)
+        .inScope(BindingScope.SINGLETON);
+
+      let value = await invokeMethod(Store, 'getHash', requestCtx);
```

Comment:
Are these two lines (and 257-258) showing how it can be done before the feature to inject method parameters down the context chain?

Reply:
It verifies that method parameter injections for singleton are still allowed.